selftests/functional/test_basic.py

```diff
@@ -269,7 +269,7 @@ def test_unsupported_status(self):
                               " --json -" % (AVOCADO, self.tmpdir, tst),
                               ignore_status=True)
             self.assertEqual(res.exit_status, exit_codes.AVOCADO_TESTS_FAIL)
-            results = json.loads(res.stdout)
+            results = json.loads(res.stdout_text)
```

Comment:
Ok. Both Python 2.7 and 3.6 accept bytes. If we claim "Python3" compatibility, then we have to make it work in 3.4 as well.